js/course.js

```diff
@@ -92,6 +92,26 @@ function setCourseOptionsContent(modal, options) {
             }
         }
     }
+
+    // Add request icon button only if default icon doesn't already exist
+    let hasDefaultIcon = (function (course) {

```

Comment:
This logic should be factored out, since we use it more than once and it's slightly nontrivial. Maybe a static method on `Theme`? E.g. I'm worried about if we need to change use of `.` as the default-indicating regex, or if we add broad auto-assumed icons (e.g. using LAUSD's course list as suggested in the issue) and want to change the behavior here to more accurately reflect what's a "real" icon or not.